OpenRA.Mods.Common/Traits/World/AutoSave.cs

```diff
@@ -0,0 +1,86 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[TraitLocation(SystemActors.World)]
+	[Desc("Add this trait to the world actor to enable auto-save.")]
+	public class AutoSaveInfo : TraitInfo
+	{
+		public override object Create(ActorInitializer init) { return new AutoSave(init.Self, this); }
+	}
+
+	public class AutoSave : ITick
+	{
+		const string AutoSavePattern = "auto-save@";
+		const string SaveFileExtension = ".orasav";
+		readonly DirectoryInfo autoSaveDirectoryInfo;
+		readonly string saveFolderPath;
+		uint ticksUntilAutoSave;
+
+		public AutoSave(Actor self, AutoSaveInfo info)
+		{
+			ticksUntilAutoSave = (uint)GetTicksBetweenAutosaves(self);
+
+			var mod = Game.ModData.Manifest;
+
+			saveFolderPath = Path.Combine(Platform.SupportDir, "Saves", mod.Id, mod.Metadata.Version);
+
+			autoSaveDirectoryInfo = new DirectoryInfo(saveFolderPath);
+		}
+
+		void ITick.Tick(Actor self)
+		{
+			var autoSaveValue = Game.Settings.Gameplay.AutoSaveInterval;
+			var autoSaveFileLimit = Game.Settings.Gameplay.AutoSaveFileNumber;
+
+			if (self.World.Paused || autoSaveValue == 0 || autoSaveFileLimit == 0) return;
+
+			if (ticksUntilAutoSave == 0)
+				ticksUntilAutoSave = (uint)GetTicksBetweenAutosaves(self);
+
+			if (--ticksUntilAutoSave > 0)
+				return;
+
+			if (GetAutoSaveFiles().Count() >= autoSaveFileLimit)
+			{
+				var oldAutoSaveFiles = GetAutoSaveFiles()
+					.OrderByDescending(f => f.CreationTime)
+					.Skip(autoSaveFileLimit - 1)
+					.ToList();
+				oldAutoSaveFiles.ForEach(f => f.Delete());
+			}
```

Comment:
```suggestion
			if (autoSaveFileLimit > 0)
			{
				foreach (var file in GetAutoSaveFiles()
					.OrderByDescending(f => f.CreationTime)
					.Skip(autoSaveFileLimit - 1)
					.ToList())
				{
					file.Delete();
				}
			}
```

Reply:
If there is no limit, we should just disabled the deleting function instead of disabling autosaves

Reply:
@PunkPun if there is no limit, it will not reach that area of code